### PR TITLE
define 'allocation' and 'memory address'

### PR DESCRIPTION
**Pull request by RalfJung**
Fixes https://github.com/rust-lang/unsafe-code-guidelines/issues/213

### CODE REVIEW

File: reference/src/glossary.md [hunk 1]
```diff
@@ -119,6 +136,12 @@ Note that layout does not capture everything that there is to say about how a ty
 Note: Originally, *layout* and *representation* were treated as synonyms, and Rust language features like the `#[repr]` attribute reflect this. 
 In this document, *layout* and [*representation*][representation relation] are not synonyms.
 
+### Memory Address
+[memory address]: #memory-address
+
+A *memory address* is an integer value that identifies where in the physical machine memory some data is stored.
```
Comment (mkroening) [inline reference/src/glossary.md:142]: Bare-metal programmers may confuse this with virtual/physical memory addresses from paging.

Is the term physical machine already established in this context?

Comment (RalfJung) [reply reference/src/glossary.md]: Ah right, I meant "physical" in contrast to the Abstract Machine, but there's always more layers of abstractions further down the stack. I adjusted the wording, is it better now?

Comment (mkroening) [reply reference/src/glossary.md]: Great, thank you! :)

**Review by saethlin — APPROVED** [review]

**Comment by RalfJung** [comment]
No objections, one approval -- I'll land this, please file an issue or a PR if you disagree with anything. :)

**Pull request merged**